src/dynimage.rs

```diff
@@ -878,30 +878,81 @@ impl DynamicImage {
         dynamic_map!(*self, ref p => imageops::flip_vertical(p))
     }
 
+    /// Flip this image vertically in place
+    pub fn flipv_in_place(&mut self) {
+        dynamic_map!(*self, ref mut p, imageops::flip_vertical_in_place(p))
+    }
+
     /// Flip this image horizontally
     #[must_use]
     pub fn fliph(&self) -> DynamicImage {
         dynamic_map!(*self, ref p => imageops::flip_horizontal(p))
     }
 
+    /// Flip this image horizontally in place
+    pub fn fliph_in_place(&mut self) {
+        dynamic_map!(*self, ref mut p, imageops::flip_horizontal_in_place(p))
+    }
+
     /// Rotate this image 90 degrees clockwise.
     #[must_use]
     pub fn rotate90(&self) -> DynamicImage {
         dynamic_map!(*self, ref p => imageops::rotate90(p))
     }
 
-    /// Rotate this image 180 degrees clockwise.
+    /// Rotate this image 180 degrees.
     #[must_use]
     pub fn rotate180(&self) -> DynamicImage {
         dynamic_map!(*self, ref p => imageops::rotate180(p))
     }
 
+    /// Rotate this image 180 degrees in place.
+    pub fn rotate180_in_place(&mut self) {
+        dynamic_map!(*self, ref mut p, imageops::rotate180_in_place(p))
+    }
+
     /// Rotate this image 270 degrees clockwise.
     #[must_use]
     pub fn rotate270(&self) -> DynamicImage {
         dynamic_map!(*self, ref p => imageops::rotate270(p))
     }
 
+    /// Applies the [Exif orientation](https://web.archive.org/web/20200412005226/https://www.impulseadventure.com/photo/exif-orientation.html) to the image.
+    ///
+    /// Orientation is specified in the Exif metadata, and is often written by cameras.
+    /// It is expressed as an integer in the range 1..=8; passing other values will return an error.
+    ///
+    /// Due to an implementation detail, orientations 5..=8 copy the image internally.
+    pub fn apply_exif_orientation_in_place(&mut self, orientation: u8) -> Result<(), ImageError> {
```

Comment:
I'd rather this function take an `Orientation` enum instead of a `u8`. That would have a couple nice benefits:

- This function would become infallible and thus wouldn't need to return a `Result`.
- We'd no longer need to add the standalone `*_in_place` methods since we could just recommend that users instead call this one with a hard-coded value.
- We could reuse the enum if we added a method to extract the orientation from an EXIF chunk (full EXIF parsing feels out of scope this crate, but adding logic to extract a single field feels pretty manageable)

Reply:
Sounds good! I'll update the PR with these changes.